toonz/sources/toonz/startuppopup.cpp

```diff
@@ -0,0 +1,889 @@
+                       SLOT(updateResolution()));
+  ret = ret && connect(m_heightFld, SIGNAL(valueChanged()), this,
+                       SLOT(updateResolution()));
+  ret = ret &&
+        connect(m_resXFld, SIGNAL(valueChanged()), this, SLOT(updateSize()));
+  ret = ret &&
+        connect(m_resYFld, SIGNAL(valueChanged()), this, SLOT(updateSize()));
+  ret = ret && connect(m_dpiFld, SIGNAL(editingFinished()), this,
+                       SLOT(onDpiChanged()));
+  ret = ret && connect(m_presetCombo, SIGNAL(activated(const QString &)),
+                       SLOT(onPresetSelected(const QString &)));
+  ret = ret && connect(m_addPresetBtn, SIGNAL(clicked()), SLOT(addPreset()));
+  ret = ret && connect(m_unitsCB, SIGNAL(currentIndexChanged(int)),
+                       SLOT(onCameraUnitChanged(int)));
+  ret = ret &&
+        connect(m_removePresetBtn, SIGNAL(clicked()), SLOT(removePreset()));
+
+  assert(ret);
+}
+
+//-----------------------------------------------------------------------------
+
+void StartupPopup::showEvent(QShowEvent *) {
+  loadPresetList();
+  updateProjectCB();
+  m_nameFld->setFocus();
+  m_pathFld->setPath(TApp::instance()
+                         ->getCurrentScene()
+                         ->getScene()
+                         ->getProject()
+                         ->getScenesPath()
+                         .getQString());
+  TDimensionD cameraSize = TApp::instance()
+                               ->getCurrentScene()
+                               ->getScene()
+                               ->getCurrentCamera()
+                               ->getSize();
+  TDimension cameraRes = TApp::instance()
+                             ->getCurrentScene()
+                             ->getScene()
+                             ->getCurrentCamera()
+                             ->getRes();
+  double fps = TApp::instance()
+                   ->getCurrentScene()
+                   ->getScene()
+                   ->getProperties()
+                   ->getOutputProperties()
+                   ->getFrameRate();
+  m_widthFld->setValue(cameraSize.lx);
+  m_heightFld->setValue(cameraSize.ly);
+  if (Preferences::instance()->getCameraUnits() == "pixel") {
+    m_widthFld->setDecimals(0);
+    m_heightFld->setDecimals(0);
+    m_resTextLabel->hide();
+    m_resXFld->hide();
+    m_resYFld->hide();
+    m_resXLabel->hide();
+    m_dpiFld->hide();
+    m_dpiLabel->hide();
+  } else {
+    m_widthFld->setDecimals(4);
+    m_heightFld->setDecimals(4);
+    m_resXFld->show();
+    m_resYFld->show();
+    m_resXLabel->show();
+    m_resTextLabel->show();
+    m_dpiFld->show();
+    m_dpiLabel->show();
+  }
+
+  m_fpsFld->setValue(fps);
+  m_unitsCB->setCurrentText(Preferences::instance()->getCameraUnits());
+  m_dpi  = cameraRes.lx / cameraSize.lx;
+  m_xRes = cameraRes.lx;
+  m_yRes = cameraRes.ly;
+  m_resXFld->setValue(m_xRes);
+  m_resYFld->setValue(m_yRes);
+  m_resXFld->setDecimals(0);
+  m_resYFld->setDecimals(0);
+
```

Comment:
Here, `m_dpiFld` should also be initialized with `m_dpi`.
Please insert `m_dpiFld->setValue(m_dpi);`